File: Data Structures & Algorithms/search-2d-matrix/submission-2.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        // Define variables below:
        int rows = matrix.length;
        int columns = matrix[0].length;

        //First find the correct row using binary search
        int top = 0;
        int bottom = rows - 1;
        int mid = 0;

        while(top <= bottom){
            mid = (top + bottom) / 2;
            if(matrix[mid][0] <= target && matrix[mid][columns-1] >= target)
                break;
            else if(matrix[mid][0] < target)
                top = mid + 1;
            else
                bottom = mid - 1;
        }

        int row = mid;
        System.out.println("Search row is "+ row);
        //Search within the correct column using binary search

        int left = 0;
        int right = columns - 1;

        while(left <= right){
            mid = (left + right) / 2;
            if(matrix[row][mid] == target)
                return true;
            else if(matrix[row][mid] > target)
                right = mid - 1;
            else
                left = mid + 1;
        }


        return false;
    }
}

File: Data Structures & Algorithms/find-k-closest-elements/submission-2.py
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        # arr = [2,4,5,8] k = 2, x = 6
        # output = [4,5]

        left = 0
        right = len(arr) - k

        while left < right:
            mid = (left + right) // 2
            if x - arr[mid] > arr[mid+k] - x:
                left = mid + 1
            else:
                right = mid
        
        return arr[left: left+k]
        
        